### PR TITLE
Diescalo/lib sync

### PR DESCRIPTION
**Pull request by diescalo**

### CODE REVIEW

File: library/src/main/java/com/digi/xbee/api/packet/raw/RX16IOPacket.java [hunk 1]
```diff
@@ -222,11 +229,28 @@ public int getReceiveOptions() {
...
 	 */
 	public IOSample getIOSample() {
 		return ioSample;
 	}
 	
+	/**
+	 * Returns the IO samples corresponding to the data contained in the
+	 * packet.
+	 * 
+	 * @return The IO samples of the packet, {@code null} if the packet has not
```
Comment (hgonzaleDigi) [inline library/src/main/java/com/digi/xbee/api/packet/raw/RX16IOPacket.java]: NP: "if the packet does not have any data" sounds better

File: library/src/main/java/com/digi/xbee/api/packet/ip/TXTLSProfilePacket.java [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+import com.digi.xbee.api.packet.APIFrameType;
+import com.digi.xbee.api.packet.XBeeAPIPacket;
+import com.digi.xbee.api.utils.HexUtils;
+
+/**
+ * This class represents a TX (Transmit) request with TLS profile packet.
+ * The packet implies the use of TLS and behaves similar to the TX Request
+ * (0x20) frame, with the protocol field replaced with a TLS Profile field to
+ * choose from the profiles configured with the $0, $1, and $2 configuration 
+ * commands.Packet is built using the parameters of the constructor or
```
Comment (hgonzaleDigi) [inline library/src/main/java/com/digi/xbee/api/packet/ip/TXTLSProfilePacket.java]: NP: missing whitespace after colon.

File: library/src/main/java/com/digi/xbee/api/packet/ip/TXTLSProfilePacket.java [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+
+	private byte[] data;
+
+	private Logger logger;
+
+	/**
+	 * Creates a new {@code TXTLSProfilePacket} object from the given payload.
+	 *
+	 * @param payload The API frame payload. It must start with the frame type
+	 *                corresponding to a TX TX request with TLS profile packet
```
Comment (rubenmoral) [inline library/src/main/java/com/digi/xbee/api/packet/ip/TXTLSProfilePacket.java]: NP: duplicated TX.

File: library/src/main/java/com/digi/xbee/api/models/HardwareVersionEnum.java [hunk 4]
```diff
@@ -95,17 +95,21 @@
...
-	// TODO: Add the since annotation in the next version of the library.
+	CELLULAR_NBIOT_EUROPE(0x48, "XBee Cellular NBIoT Europe"),  // Never released
+	/** @since 1.3.0 */
 	CELLULAR_3_CAT1_LTE_ATT(0x49, "XBee Cellular 3 Cat 1 LTE AT&T"),
-	// TODO: Add the since annotation in the next version of the library.
-	CELLULAR_3_LTE_M_VERIZON(0x4A, "XBee Cellular 3 LTE-M Verizon"),
-	// TODO: Add the since annotation in the next version of the library.
-	CELLULAR_3_LTE_M_ATT(0x4B, "XBee Cellular 3 LTE-M AT&T");
+	/** @since 1.3.0 */
+	CELLULAR_3_LTE_M_VERIZON(0x4A, "XBee Cellular 3 LTE-M Verizon"),  // Abandoned
```
Comment (rubenmoral) [inline library/src/main/java/com/digi/xbee/api/models/HardwareVersionEnum.java]: I would use "XBee3 Cellular [...]" as name.

**Pull request merged**